FrameworksAndDrivers/Web/DotLanches.Api/Controllers/PagamentoController.cs

```diff
@@ -25,19 +25,35 @@ public PagamentoController(IPagamentoRepository pagamentoRepository, IPedidoRepo
         }
 
         /// <summary>
-        /// Envia os dados para o meio de pagamento e retorna a senha para retirar o pedido.
+        /// Envia os dados para o meio de pagamento e retorna o QR Code para o cliente.
         /// </summary>
         /// <param name="pagamentoRequest">Dados de requisição do pagamento</param>
+        /// <returns>QR Code para efetuar pagamento.</returns>
+        [HttpPost("QrCode")]
+        [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> ExecutePayment([Required][FromBody] PagamentoRequestDto pagamentoRequest)
```

Comment:
Esse endpoint agora está apenas requisitando o qr code e retornando, certo? Acredito que podemos atualizar o nome para deixar mais claro o que está fazendo:

```suggestion
        public async Task<IActionResult> RequestPagamentoQrCode([Required][FromBody] PagamentoRequestDto pagamentoRequest)
```